Description: Update FindRuby.cmake to also support Ruby 2.0 and 2.1.
 The most significant change since Ruby 1.9 is the changed path of
 config.h, from RbConfig::CONFIG['rubyhdrdir'] to
 RbConfig::CONFIG['rubyarchhdrdir'].
Author: Bas Couwenberg <sebastic@xs4all.nl>
Forwarded: https://github.com/Kitware/CMake/pull/92
Bug-Debian: https://bugs.debian.org/739826
Applied-Upstream: https://github.com/Kitware/CMake/commit/165f6430d957173a0c3fda48265d8ed761700e20
--- a/Modules/FindRuby.cmake
+++ b/Modules/FindRuby.cmake
@@ -1,20 +1,32 @@
-# - Find Ruby
-# This module finds if Ruby is installed and determines where the include files
-# and libraries are. Ruby 1.8 and 1.9 are supported.
+#.rst:
+# FindRuby
+# --------
+#
+# Find Ruby
+#
+# This module finds if Ruby is installed and determines where the
+# include files and libraries are.  Ruby 1.8, 1.9, 2.0 and 2.1 are
+# supported.
 #
 # The minimum required version of Ruby can be specified using the
-# standard syntax, e.g. find_package(Ruby 1.8)
+# standard syntax, e.g.  find_package(Ruby 1.8)
 #
-# It also determines what the name of the library is. This
-# code sets the following variables:
+# It also determines what the name of the library is.  This code sets
+# the following variables:
 #
-#  RUBY_EXECUTABLE   = full path to the ruby binary
-#  RUBY_INCLUDE_DIRS = include dirs to be used when using the ruby library
-#  RUBY_LIBRARY      = full path to the ruby library
-#  RUBY_VERSION      = the version of ruby which was found, e.g. "1.8.7"
-#  RUBY_FOUND        = set to true if ruby ws found successfully
+# ::
 #
-#  RUBY_INCLUDE_PATH = same as RUBY_INCLUDE_DIRS, only provided for compatibility reasons, don't use it
+#   RUBY_EXECUTABLE   = full path to the ruby binary
+#   RUBY_INCLUDE_DIRS = include dirs to be used when using the ruby library
+#   RUBY_LIBRARY      = full path to the ruby library
+#   RUBY_VERSION      = the version of ruby which was found, e.g. "1.8.7"
+#   RUBY_FOUND        = set to true if ruby ws found successfully
+#
+#
+#
+# ::
+#
+#   RUBY_INCLUDE_PATH = same as RUBY_INCLUDE_DIRS, only provided for compatibility reasons, don't use it
 
 #=============================================================================
 # Copyright 2004-2009 Kitware, Inc.
@@ -56,6 +68,8 @@ else()
 endif()
 
 if(NOT Ruby_FIND_VERSION_EXACT)
+  list(APPEND _RUBY_POSSIBLE_EXECUTABLE_NAMES ruby2.1 ruby21)
+  list(APPEND _RUBY_POSSIBLE_EXECUTABLE_NAMES ruby2.0 ruby20)
   list(APPEND _RUBY_POSSIBLE_EXECUTABLE_NAMES ruby1.9 ruby19)
 
   # if we want a version below 1.9, also look for ruby 1.8
@@ -94,6 +108,7 @@ if(RUBY_EXECUTABLE  AND NOT  RUBY_VERSIO
    _RUBY_CONFIG_VAR("archdir" RUBY_ARCH_DIR)
    _RUBY_CONFIG_VAR("arch" RUBY_ARCH)
    _RUBY_CONFIG_VAR("rubyhdrdir" RUBY_HDR_DIR)
+   _RUBY_CONFIG_VAR("rubyarchhdrdir" RUBY_ARCHHDR_DIR)
    _RUBY_CONFIG_VAR("libdir" RUBY_POSSIBLE_LIB_DIR)
    _RUBY_CONFIG_VAR("rubylibdir" RUBY_RUBY_LIB_DIR)
 
@@ -115,7 +130,8 @@ if(RUBY_EXECUTABLE  AND NOT  RUBY_VERSIO
    set(RUBY_VERSION_MINOR    ${RUBY_VERSION_MINOR}    CACHE PATH "The Ruby minor version" FORCE)
    set(RUBY_VERSION_PATCH    ${RUBY_VERSION_PATCH}    CACHE PATH "The Ruby patch version" FORCE)
    set(RUBY_ARCH_DIR         ${RUBY_ARCH_DIR}         CACHE PATH "The Ruby arch dir" FORCE)
-   set(RUBY_HDR_DIR          ${RUBY_HDR_DIR}          CACHE PATH "The Ruby header dir (1.9)" FORCE)
+   set(RUBY_HDR_DIR          ${RUBY_HDR_DIR}          CACHE PATH "The Ruby header dir (1.9+)" FORCE)
+   set(RUBY_ARCHHDR_DIR      ${RUBY_ARCHHDR_DIR}      CACHE PATH "The Ruby arch header dir (2.0+)" FORCE)
    set(RUBY_POSSIBLE_LIB_DIR ${RUBY_POSSIBLE_LIB_DIR} CACHE PATH "The Ruby lib dir" FORCE)
    set(RUBY_RUBY_LIB_DIR     ${RUBY_RUBY_LIB_DIR}     CACHE PATH "The Ruby ruby-lib dir" FORCE)
    set(RUBY_SITEARCH_DIR     ${RUBY_SITEARCH_DIR}     CACHE PATH "The Ruby site arch dir" FORCE)
@@ -128,6 +144,7 @@ if(RUBY_EXECUTABLE  AND NOT  RUBY_VERSIO
      RUBY_ARCH_DIR
      RUBY_ARCH
      RUBY_HDR_DIR
+     RUBY_ARCHHDR_DIR
      RUBY_POSSIBLE_LIB_DIR
      RUBY_RUBY_LIB_DIR
      RUBY_SITEARCH_DIR
@@ -149,10 +166,20 @@ if(RUBY_EXECUTABLE AND NOT RUBY_VERSION_
    set(RUBY_VERSION_MINOR 8)
    set(RUBY_VERSION_PATCH 0)
    # check whether we found 1.9.x
-   if(${RUBY_EXECUTABLE} MATCHES "ruby1.?9"  OR  RUBY_HDR_DIR)
+   if(${RUBY_EXECUTABLE} MATCHES "ruby1.?9")
       set(RUBY_VERSION_MAJOR 1)
       set(RUBY_VERSION_MINOR 9)
    endif()
+   # check whether we found 2.0.x
+   if(${RUBY_EXECUTABLE} MATCHES "ruby2.?0")
+      set(RUBY_VERSION_MAJOR 2)
+      set(RUBY_VERSION_MINOR 0)
+   endif()
+   # check whether we found 2.1.x
+   if(${RUBY_EXECUTABLE} MATCHES "ruby2.?1")
+      set(RUBY_VERSION_MAJOR 2)
+      set(RUBY_VERSION_MINOR 1)
+   endif()
 endif()
 
 if(RUBY_VERSION_MAJOR)
@@ -178,6 +205,7 @@ if( "${Ruby_FIND_VERSION_SHORT_NODOT}" G
      HINTS
      ${RUBY_HDR_DIR}/${RUBY_ARCH}
      ${RUBY_ARCH_DIR}
+     ${RUBY_ARCHHDR_DIR}
      )
 
    set(RUBY_INCLUDE_DIRS ${RUBY_INCLUDE_DIRS} ${RUBY_CONFIG_INCLUDE_DIR} )
